client/signup/main.jsx

```diff
@@ -738,7 +740,7 @@ class Signup extends React.Component {
 				{ ! isWPForTeamsFlow( this.props.flowName ) && (
 					<SignupHeader
 						positionInFlow={ this.getPositionInFlow() }
-						flowLength={ this.getFlowLength() }
+						flowLength={ this.getInteractiveStepsCount() }
```

Comment:
It seems the only component that uses `flowLength`. Maybe we should be move this call to there. But then we'll need to drill isLogged in 🤷🏼

Reply:
I was thinking of that too but `FlowProgressIndicator` is currently a presentational component and it may be as well extracted to `components`. 

Then, a possible alternative would be to connect `SignupHeader` to the store to get `isLoggedIn` there. 

But probably the best solution here would be to render `FlowProgressIndicator` directly from `Main` so we stop drilling down `showProgressIndicator`, `positionInFlow`, `flowLength`, and `flowName` props. So [these lines](https://github.com/Automattic/wp-calypso/blob/a6cb8ad2c61fd9419c4e3af760c98032fcd5ed93/client/signup/main.jsx#L741-L748) may change to: 
```
<SignupHeader
	shouldShowLoadingScreen={ this.state.shouldShowLoadingScreen }
	isReskinned={ isReskinned }
	rightComponent={
		<FlowProgressIndicator
			positionInFlow={ this.getPositionInFlow() }
			flowLength={ this.getInteractiveStepsCount() }
			flowName={ this.props.flowName }
			showProgressIndicator={ showProgressIndicator }
		/>
	}
/>
```

Reply:
Tried it in https://github.com/Automattic/wp-calypso/pull/54318/commits/be074b7dcf42a59ca4a6ee15dfc2332dc7f927bc and looks good to me.